### PR TITLE
1984 std allocator

### PR DESCRIPTION
**Pull request by mariusztrela**
#1984

### CODE REVIEW

**Comment by mvandeberg** [comment]
Merge conflicts, please rebase on develop.

**Comment by mariusztrela** [comment]
Done.

File: libraries/chain/include/steem/chain/hardfork_property_object.hpp [hunk 1]
```diff
@@ -6,19 +6,23 @@
...
          {
             c( *this );
          }
 
          id_type                                                              id;
 
-         bip::vector< fc::time_point_sec, allocator< fc::time_point_sec > >   processed_hardforks;
+         using t_beneficiaries = t_vector< fc::time_point_sec >;
+
+         t_beneficiaries                                                      processed_hardforks;
```
Comment (mvandeberg) [inline libraries/chain/include/steem/chain/hardfork_property_object.hpp]: Typedef this name `t_beneficiaries` for processed hardforks? Naming makes no sense.

**Comment by mariusztrela** [comment]
- improper name of type ( copy-paste effect )
 - Currently `std-allocator` is disabled by default. This implementation is necessary for our further development, so if you haven't any objections, please merge this branch to `develop`.

**Comment by vogel76** [comment]
We did dedicated test (by running reindex process up to current head block). It passed and I don't see any reason to blocking merge of this PR  more.

**Comment by mvandeberg** [comment]
Doing a local reindex to confirm expected behavior.

**Comment by mariusztrela** [comment]
Is everything ok with testing this implementation?

**Comment by mvandeberg** [comment]
Everything looks good. As I mentioned to @vogel76 in our sprint meeting last week, I am waiting for the appbase release to merge this. I am pretty certain this branch is good as is, but I do not want to risk the pending appbase release if this branch gives us problems.

**Pull request merged**